### DISCUSSION TITLE
Templating Component: “This package is deprecated.”

### DISCUSSION THREAD

**Discussion by alexisLefebvre** [symfony/symfony#59002] [Q&A]
On this page: https://symfony.com/components/Templating

Wee see : 

> Templating Component
> **This package is deprecated.** Don't use it in new applications or projects.

I think that it would be nice to add that `symfony/twig-bundle` removes the need of `symfony/templating`.

If that's wrong, then it should at least explain what users have to do, giving no solution or guidance is a bit confusing.

**Comment by GromNaN** [comment]
The package `symfony/templating` is not requires by any of the other Symfony packages since version 5.0.

Run `composer why symfony/templating` to see why this package is installed.

**Reply by alexisLefebvre** [reply]
Thanks for your answer, my question may lack details so here is the full story:

I updated a project from Symfony 6.4 to 7.1, so I replaced `"symfony/templating": "^6.4.13",` with `"symfony/templating": "^7.1",` like the other dependencies, but this version didn’t exist.

So I searched why and found:

- https://symfony.com/components/Templating didn't help me to understand the situation
- https://github.com/symfony/templating has no message about a deprecation

I removed `symfony/templating` from the project and it worked, but it was a lucky move.

I had this dependency since the project started with Symfony 2.4 (!), and I may have missed that this package was deprecated when updating to `5.0`.

To improve DX, I suggest to add a message more explicit, something like:

> **This package is deprecated.** Don't use it in new applications or projects. It can be removed with Symfony >= 5.0.
